File: core/src/test/java/hello/core/member/MemberServiceTest.java
package hello.core.member;

import hello.core.AppConfig;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

public class MemberServiceTest {
    private MemberService memberService;

//    @BeforeEach
//    void setUp() {
//        AppConfig appConfig = new AppConfig();
//        memberService = appConfig.memberService();
//    }

    @BeforeEach
    void setUpByDiContainer() {
        ApplicationContext applicationContext = new AnnotationConfigApplicationContext(AppConfig.class);
        memberService = applicationContext.getBean("memberService", MemberService.class);
    }

    @Test
    void join() {
        // given
        Member member = new Member(1L, "memberA", Grade.VIP);
        // when
        memberService.join(member);
        Member findMember = memberService.findMember(1L);

        // then
        Assertions.assertThat(member).isEqualTo(findMember);
    }
}

File: src/test/kotlin/hello/core/member/MemberServiceTest.kt
package hello.core.member

import hello.core.AppConfig
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

private val appConfig = AppConfig()

internal class MemberServiceTest {
    var memberService: MemberService = appConfig.memberService()

    @Test
    fun Join() {
        // given
        val member = Member(1L, "memberA", Grade.VIP)

        // when
        memberService.join(member)
        val findMember = memberService.findMember(1L)

        // then
        assertThat(member).isEqualTo(findMember)
    }
}